python/ray/tune/ray_trial_executor.py

```diff
@@ -944,6 +877,150 @@ def _change_working_directory(self, trial):
         else:
             yield
 
+    def get_next_executor_event(
+        self, live_trials: Set[Trial], next_trial_exists: bool
+    ) -> ExecutorEvent:
+        """Get the next executor event to be processed in TrialRunner.
```

Comment:
Main fundamental question - can you help me understand why this function lives in the trial executor, as compared to the trial runner?

Reply:
TrialRunner's responsibility is to run event loop, maintain `live_trial` queue and handle all the `ExecutorEvent` (as defined in this PR). TrialExecutor is the emitter of these `ExecutorEvent`.